utils/util.go

```diff
@@ -51,3 +55,123 @@ func RemoveCellOutputData(cellOutputData [][]byte, index int) [][]byte {
+	major, minor, _, err := ParseNodeVersion(client, ctx)
+	if err != nil {
+		return nil, err
+	}
+	if major > 0 || minor >= 39 {
+		consensus, err := client.GetConsensus(ctx)
+		if err != nil {
+			return nil, err
+		}
+		cellbaseMaturity = types.ParseEpoch(consensus.CellbaseMaturity)
+	} else {
+		cellbaseMaturity = &types.EpochParams{
+			Length: 1,
+			Index:  0,
+			Number: 4,
+		}
+	}
+	return cellbaseMaturity, nil
+}
+
+// startNumber is maxMatureEpoch.StartNumber, length is maxMatureEpoch.Length
+func calcMaxMatureBlockNumber(tipEpoch *types.EpochParams, startNumber, length uint64, cellbaseMaturity *types.EpochParams) (uint64, error) {
+	a := isTipEpochLessThanCellbaseMaturity(tipEpoch, cellbaseMaturity)
+	if a {
+		return 0, nil
+	} else {
+		tipEpochRN, cellbaseMaturityRN := epochRationalNum(tipEpoch, cellbaseMaturity)
+		epochDelta := rationalNumber{
+			numer: tipEpochRN.numer - cellbaseMaturityRN.numer,
+			denom: tipEpochRN.denom,
+		}
+		iNum := epochDelta.numer / epochDelta.denom
+		index := (epochDelta.numer - iNum*epochDelta.denom) * length / epochDelta.denom
+		blockNumber := index + startNumber
+
+		return blockNumber, nil
+	}
+}
+
+type rationalNumber struct {
+	// Numerator.
+	numer uint64
+	// Denominator.
+	denom uint64
+}
+
+func epochRationalNum(tipEpoch, cellbaseMaturity *types.EpochParams) (rationalNumber, rationalNumber) {
+	if tipEpoch.Length != cellbaseMaturity.Length {
+		tipEpochMN := rationalNumber{
+			numer: tipEpoch.Number*tipEpoch.Length + tipEpoch.Index,
+			denom: tipEpoch.Length,
+		}
+		cellbaseMaturityMN := rationalNumber{
+			numer: (cellbaseMaturity.Number*cellbaseMaturity.Length + cellbaseMaturity.Index) * tipEpoch.Length,
+			denom: cellbaseMaturity.Length * cellbaseMaturity.Length * tipEpoch.Length,
+		}
+		return tipEpochMN, cellbaseMaturityMN
+	} else {
+		tipEpochMN := rationalNumber{
+			numer: tipEpoch.Number*tipEpoch.Length + tipEpoch.Index,
+			denom: tipEpoch.Length,
+		}
+		cellbaseMaturityMN := rationalNumber{
+			numer: cellbaseMaturity.Number*cellbaseMaturity.Length + cellbaseMaturity.Index,
+			denom: cellbaseMaturity.Length,
+		}
+		return tipEpochMN, cellbaseMaturityMN
+	}
+}
+
+func isTipEpochLessThanCellbaseMaturity(tipEpoch *types.EpochParams, cellbaseMaturity *types.EpochParams) bool {
+	tipEpochRN, cellbaseMaturityRN := epochRationalNum(tipEpoch, cellbaseMaturity)
+	if tipEpochRN.numer-cellbaseMaturityRN.numer < 0 {
+		return true
+	}
+	return false
+}
+
+// ParseNodeVersion return ckb node version number
+func ParseNodeVersion(client rpc.Client, ctx context.Context) (int, int, int, error) {
+	nodeInfo, err := client.LocalNodeInfo(ctx)
+	if err != nil {
+		return 0, 0, 0, err
+	}
+	nodeVersion := strings.Split(nodeInfo.Version, " (")
+	versionArr := strings.Split(nodeVersion[0], ".")
+	major, err := strconv.Atoi(versionArr[0])
+	minor, err := strconv.Atoi(versionArr[1])
+	patch, err := strconv.Atoi(versionArr[2])
+	return major, minor, patch, nil
```

Comment:
```suggestion
	nodeVersion := strings.Split(nodeInfo.Version, " (")
	versionArr := strings.Split(nodeVersion[0], ".")
	major, err := strconv.Atoi(versionArr[0])
	minor, err := strconv.Atoi(versionArr[1])
	patch, err := strconv.Atoi(versionArr[2])
	return major, minor, patch, nil
```
Put above line in a standalone function and test it.

Reply:
updated: https://github.com/nervosnetwork/ckb-sdk-go/pull/26/commits/02abcd57065ec513f5865bbf86f72397059e2fa6